scriptmodules/supplementary/esthemes.sh

```diff
@@ -24,15 +24,19 @@ function depends_esthemes() {
 function install_theme_esthemes() {
     local theme="$1"
     local repo="$2"
+    local default_branch
     if [[ -z "$repo" ]]; then
         repo="RetroPie"
     fi
     if [[ -z "$theme" ]]; then
         theme="carbon"
         repo="RetroPie"
     fi
+    # Get the name of the default branch, fallback to 'master' if not found
+    default_branch=$(runCmd git ls-remote --symref "$repo" HEAD | cut -s -d'/' -f3 | cut -d$'\t' -f1)
```

Comment:
I think the two cuts in this case are maybe less readable than a regexp via grep or sed. Eg

```
 | grep -oP ".*/\K\w+"
```

Although I feel there should be something simpler still maybe.

Reply:
Does `| awk -F'[\t/]' '/ref/ { print $(NF-1); exit}' ` look better ? It's not shorter, but easier to read.

Reply:
Think I still prefer the grep in this case. Actually could use bash regular expressions too. I guess it's not important as it's not a performance thing but I would guess the grep is most efficient. (despite using perl regexp and so not being portable). Awk is readable if you know awk - I like it but it's long and seems a bit basic for an awk task :-)

Reply:
OK, added the `grep` filter for the default branch name.